modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -98,6 +109,9 @@ public QueryMemoryManager(GridKernalContext ctx, long globalQuota) {
 
         this.blockSize = Long.getLong(IgniteSystemProperties.IGNITE_SQL_MEMORY_RESERVATION_BLOCK_SIZE, DFLT_MEMORY_RESERVATION_BLOCK_SIZE);
         this.globalQuota = globalQuota;
+        // TODO GG-18629 - get from configuration.
+        this.failOnMemLimitExceed = Boolean.getBoolean(IgniteSystemProperties.IGNITE_SQL_FAIL_ON_QUERY_MEMORY_LIMIT_EXCEED);
```

Comment:
Would IGNITE_SQL_USE_DISK_OFFLOAD be more clear to user?

Reply:
Agree. This name looks more meaningful. Done.